sdk/eventhub/azure-eventhubs/azure/eventhub/receiver.py

```diff
@@ -129,33 +149,40 @@ def __next__(self):
                 self.close(exception=error)
                 raise error
 
+    def _check_closed(self):
+        if self.error:
+            raise EventHubError("This receiver has been closed. Please create a new receiver to receive event data.",
+                                self.error)
+
+    def _redirect(self, redirect):
+        self.redirected = redirect
+        self.running = False
+        self._open()
+
     def _open(self):
         """
-        Open the Receiver using the supplied conneciton.
+        Open the EventReceiver using the supplied connection.
         If the handler has previously been redirected, the redirect
         context will be used to create a new handler before opening it.
 
-        :param connection: The underlying client shared connection.
-        :type: connection: ~uamqp.connection.Connection
-
-        Example:
-            .. literalinclude:: ../examples/test_examples_eventhub.py
-                :start-after: [START eventhub_client_receiver_open]
-                :end-before: [END eventhub_client_receiver_open]
-                :language: python
-                :dedent: 4
-                :caption: Open the Receiver using the supplied conneciton.
-
         """
         # pylint: disable=protected-access
+        self._check_closed()
         if self.redirected:
+            self.client._process_redirect_uri(self.redirected)
             self.source = self.redirected.address
             source = Source(self.source)
             if self.offset is not None:
-                source.set_filter(self.offset.selector())
+                source.set_filter(self.offset._selector())
+
             alt_creds = {
                 "username": self.client._auth_config.get("iot_username"),
                 "password":self.client._auth_config.get("iot_password")}
+            '''
+            alt_creds = {
+                "username": self.client._auth_config.get("username"),
+                "password": self.client._auth_config.get("password")}
+            '''
```

Comment:
Duplicate?

Reply:
This is commented out debugging trash code. Removed now